internal/pkg/saelastic/client.go

```diff
@@ -80,6 +80,10 @@ func CreateClient(config saconfig.EventConfiguration) (*ElasticClient, error) {
 		elasticOpts = append(elasticOpts, elastic.SetHttpClient(tlsClient), elastic.SetScheme("https"))
 	}
 
+	if config.UseBasicAuth {
+		elasticOpts = append(elasticOpts, elastic.SetBasicAuth(config.ElasticUser, config.ElasticPass))
```

Comment:
Not sure what would happen if someone provided `UseBasicAuth: true` and then didn't set the proper configurations that get appended.

Reply:
Ah yes, there is no way to set defaults for JSON parameters. It's probably worth testing, but the likely result is crashing with an error in every case (no option, bad option, good option bad ES, etc).

While refreshing my memory about defaults, though, I realized you didn't add command line flag parsing for this. Check out https://github.com/redhat-service-assurance/smart-gateway/blob/master/internal/pkg/events/events.go#L190

Reply:
We also didn't add flags for TLS configuration either...

I actually wonder if we really even need to add all of these flags, or if we should actually back most of them out. I'd prefer to load all of the configuration from a configuration file to be honest.

Reply:
Okay, I could get behind that. I already have this issue open: https://github.com/redhat-service-assurance/smart-gateway/issues/68 and dropping all command line support would certainly close it (lopping off the limb). It's a shame, though. Maybe we can stick to config-file only for now and research if there is a library that will give us a consistent low-effort interface to either/or/both.

Reply:
Yea I'm mostly fine with either setup, but it would be nice if it was consistent. I like the idea of maybe some sort of library that can auto-generate the flags for us. That'd be pretty cool.

For now it seems like we're mostly happy with this change, and I can go ahead and merge it, so I will do so. I'll open an issue about the flags and note #68 as well so that we can make a decision one way or the other.

After we get this all out for GA I want to have at least a sprint worth of effort to clean up the SG and make the code easier to maintain, test, and generally just look nicer.